specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json

```diff
@@ -0,0 +1,1234 @@
+          "enum": [
+            "TCP",
+            "UDP"
+          ],
+          "x-ms-enum": {
+            "name": "ContainerNetworkProtocol",
+            "modelAsString": true
+          }
+        },
+        "port": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The port number exposed within the container group."
+        }
+      },
+      "required": [
+        "port"
+      ]
+    },
+    "EnvironmentVariable": {
+      "description": "The environment variable to set within the container instance.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the environment variable."
+        },
+        "value": {
+          "type": "string",
+          "description": "The value of the environment variable."
+        }
+      },
+      "required": [
+        "name",
+        "value"
+      ]
+    },
+    "OperationListResult":{
+      "description": "The operation list response that contains all operations for Azure Container Instance service.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Operation"
+          },
+          "description": "The list of operations."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URI to fetch the next page of operations."
+        }
+      }
+    },
+    "Operation": {
+      "description": "An operation for Azure Container Instance service.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the operation."
+        },
+        "display": {
+          "description": "The display information of the operation.",
+          "type": "object",
+          "properties": {
+            "provider": {
+              "description": "The name of the provider of the operation.",
+              "type": "string"
+            },
+            "resource": {
+              "type": "string",
+              "description": "The name of the resource type of the operation."
+            },
+            "operation": {
+              "description": "The friendly name of the operation.",
+              "type": "string"
+            },
+            "description": {
+              "description": "The description of the operation.",
+              "type": "string"
+            }
+          }
+        },
+        "origin": {
+          "type": "string",
+          "description": "The intended executor of the operation.",
+          "enum": [
+            "User",
+            "System"
+          ],
+          "x-ms-enum": {
+            "name": "ContainerInstanceOperationsOrigin",
+            "modelAsString": true
+          }
+        }
+      },
+      "required": [
+          "name",
+          "display"
+      ]
```

Comment:
does not make sense to put required fields in the type that is used only as a server response.

Reply:
Done